extra_data/reader.py

```diff
@@ -162,6 +167,11 @@ def __eq__(self, other):
     def __repr__(self):
         return "{}({})".format(type(self).__name__, repr(self.filename))
 
+    def __getstate__(self):
```

Comment:
Let's have a brief docstring/comment to mention that this allows pickling.

Reply:
Ok